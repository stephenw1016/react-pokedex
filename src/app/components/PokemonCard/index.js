// @flow
import * as React from 'react';
import { Link } from 'react-router-dom';

type Props = {
  pokemon: { id: number, name: string }
};

export default class PokemonCard extends React.Component<Props> {
  constructor (props: Props) {
    super(props);
  }

  render (): React.Node {
    let pokemon = this.props.pokemon;

    return (
      <div className="pokemon-card">
        <div className="card-top">
          <img src={`http://res.cloudinary.com/dwnebujkh/image/upload/v1473910425/pokemon/${pokemon.id}.png`} />
        </div>
        <div className="card-bottom">
          <span>{pokemon.name}</span>
          <Link to={`/pokemon/${pokemon.id}`}>Details</Link>
        </div>
      </div>
    );
  }
}
