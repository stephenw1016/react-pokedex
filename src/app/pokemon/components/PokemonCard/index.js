// @flow
import * as React from 'react';
import { Link } from 'react-router-dom';

type Props = {
  pokemon: { id: number, name: string }
};

export default ({pokemon}: Props) => {
  let imageUrl = `http://res.cloudinary.com/dwnebujkh/image/upload/v1473910425/pokemon/${pokemon.id}.png`;
  let detailViewRoute = `/pokemon/${pokemon.id}`;

  return (
    <div className="pokemon-card">
      <div className="card-top">
        <img src={imageUrl} />
      </div>
      <div className="card-bottom">
        <span>{pokemon.name}</span>
        <Link to={detailViewRoute}>Details</Link>
      </div>
    </div>
  );
}
