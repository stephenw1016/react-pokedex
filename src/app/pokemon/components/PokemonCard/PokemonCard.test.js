import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import PokemonCard from '../PokemonCard';

test('should render pokemon', () => {
  let componentJSX = <MemoryRouter><PokemonCard pokemon={{ id: 1, name: 'bulbasaur' }} /></MemoryRouter>;
  let componentJSON = renderer.create(componentJSX).toJSON();

  expect(componentJSON).toMatchSnapshot();
});
