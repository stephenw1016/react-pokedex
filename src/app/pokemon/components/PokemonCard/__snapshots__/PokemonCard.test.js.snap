// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render pokemon 1`] = `
<div
  className="pokemon-card"
>
  <div
    className="card-top"
  >
    <img
      src="http://res.cloudinary.com/dwnebujkh/image/upload/v1473910425/pokemon/1.png"
    />
  </div>
  <div
    className="card-bottom"
  >
    <span>
      bulbasaur
    </span>
    <a
      href="/pokemon/1"
      onClick={[Function]}
    >
      Details
    </a>
  </div>
</div>
`;
